Mustachio/TokenExpander.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Mustachio
+{
+    public enum Precedence
+    {
+        Low, Medium, High
+    }
+
+    public class TokenExpander
+    {
+        /// <summary>
+        /// RegEx used to identify the custom token.
+        /// </summary>
+        public string RegEx { get; set; }
```

Comment:
Make this use the `Regex` class instead - this will ensure the regex is compiled once. Match against it instead of using "IsMatch" later, and pass the match.value to the expander directly when it is set.

If you wish, you may add constructor overloads to convert a string to the pattern for convenience.